repository/repository.go

```diff
@@ -0,0 +1,19 @@
+package repository
+
+import (
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/configuration"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var (
+	logger         *configuration.Logger
+	userCollection *mongo.Collection
+	context        = configuration.GetContext()
+)
+
+func Initialize() {
+	logger = configuration.GetLogger("repository")
+	logger.Info("Initializing repository...")
+
+	userCollection = configuration.GetUserCollection()
+}
```

Comment:
To adhere to Golang conventions and enhance modularity, consider the following refactoring using a New function.